src/test/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/HearingPartControllerTest.java

```diff
@@ -0,0 +1,145 @@
+import uk.gov.hmcts.reform.sandl.snlevents.service.RulesService;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.AdditionalAnswers.returnsFirstArg;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(HearingPartController.class)
+public class HearingPartControllerTest {
+    public static final String TYPE = "type";
+    public static final String CASE_NUMBER = "90";
+    public static final String TITLE = "title";
+    public static final String HEARING_TYPE = "hearing-type";
+    public static final String URL = "/hearing-part";
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private HearingPartService hearingPartService;
+
+    @MockBean
+    @SuppressWarnings("PMD.UnusedPrivateField")
+    private RulesService rulesService;
+
+    @MockBean
+    @SuppressWarnings("PMD.UnusedPrivateField")
+    private FactsMapper factsMapper;
+
+    private JacksonTester<CreateHearingPart> createHearingPartSerializer;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Before
+    public void setup() {
+        JacksonTester.initFields(this, objectMapper);
+    }
+
+    @Test
+    public void fetchAllHeartingParts_returnHearingPartsFromService() throws Exception {
+        val hearingParts = createHearingParts();
+        when(hearingPartService.getAllHearingParts()).thenReturn(hearingParts);
+
+        val response = mvc
+            .perform(get(URL))
+            .andExpect(status().isOk())
+            .andReturn().getResponse();
+
+        val r = objectMapper.readValue(response.getContentAsString(), new TypeReference<List<HearingPart>>(){});
+        assertThat(r).isEqualToComparingFieldByFieldRecursively(hearingParts);
+    }
+
+    private List<HearingPart> createHearingParts() {
+        return new ArrayList<>(Arrays.asList(createHearingPart()));
+    }
+
+    @Test
+    public void upsertHearingPart_savesHearingPartToService() throws Exception {
+        when(hearingPartService.save(any(HearingPart.class))).then(returnsFirstArg());
+
+        val content = createHearingPartSerializer.write(createCreateHearingPart()).getJson();
+        val response = mvc
+            .perform(put(URL).contentType(MediaType.APPLICATION_JSON_VALUE).content(content))
+            .andExpect(status().isOk())
+            .andReturn().getResponse();
+
+        val r = objectMapper.readValue(response.getContentAsString(), HearingPart.class);
```

Comment:
This conversion happens all over the tests. It's quite okay but we could create a custom matcher for jasmine that takes the entire 'response' variable, internally extracts the content and compares the value. If it's easy to do, then great, if not we can leave it as is

Reply:
check https://github.com/hmcts/snl-events/pull/78/commits/1d9469ae770bd29224aa89c6c452f7dd921fda62

It's not the best, but it does the job. Any improvement suggestions are welcome.